transports/tcp/src/main/java/org/mule/transport/tcp/TcpConnector.java

```diff
@@ -154,6 +154,18 @@ private boolean newValue(int parameter, int socketValue)
     @Override
     protected void doInitialise() throws InitialisationException
     {
+        if (tcpProtocol != null)
+        {
+            try
+            {
+                muleContext.getInjector().inject(getTcpProtocol());
```

Comment:
Because the protocol does not go into the registry but it does need to have the serializer injected anyway. So it's either this or checking for ObjectSerializerAware